src/amplitude.js

```diff
@@ -787,14 +947,23 @@ Amplitude.prototype._mergeEventsAndIdentifys = function(numEvents) {
 
   while (eventsToSend.length < numEvents) {
     var event;
+    var noIdentifys = identifyIndex >= this._unsentIdentifys.length;
+    var noEvents = eventIndex >= this._unsentEvents.length;
+
+    // case 0: no events or identifys left
+    // note this should not happen, this means we have less events and identifys than expected
+    if (noEvents && noIdentifys) {
```

Comment:
This is the case that crashed on Android SDK. It should in theory never happen, but gracefully handle if it does